src/Resources/contao/library/Contao/StringUtil.php

```diff
@@ -511,31 +511,64 @@ public static function toHtml5($strString)
 	 * @return string The converted string
 	 *
 	 * @throws \Exception If $strString cannot be parsed
+	 * @throws \InvalidArgumentException If $strString or any token in $arrData contains php code
 	 */
 	public static function parseSimpleTokens($strString, $arrData)
 	{
 		$strReturn = '';
 
-		// Remove any unwanted tags (especially PHP tags)
-		$strString = strip_tags($strString, \Config::get('allowedTags'));
+		// Do not allow php code for security reasons.
+		$containsPhp = function($value) {
+			return strpos($value, '<?php') !== false
+				|| strpos($value, '<?') !== false
+				|| strpos($value, '<%') !== false
+				|| strpos($value, '<%=') !== false
+				|| strpos($value, '<script language="php">') !== false
+				|| strpos($value, '<script language=\'php\'>') !== false;
+		};
+
+		// Validate subject string
+		if ($containsPhp($strString))
+		{
+			throw new \InvalidArgumentException('PHP Code is not allowed in subject string.');
+		}
+
+		// Validate replacement data
+		foreach ($arrData as $k => $v)
+		{
+			if ($containsPhp($v))
+			{
+				throw new \InvalidArgumentException(sprintf('PHP Code is not allowed in replacement data (Key "%s").', $k));
```

Comment:
Tried to fix it in 240c5e1d8fb52a6858afc953182b9c8899893bad. Obviously the tokenizer does not detect opening tags in PHP 7 because it's not part of the language anymore. Moreover, I guess it will not detect `<?` if short open tags are disabled but again, this is not dangerous because then it will also not interpret it in `eval()`. Unfortunately, the legacy testsuite is not run on travis and I did only test it on php7. Can you give it a shot @leofeyer and @ausi?